<?xml version="1.0"?>

<launch>
      <arg name="robot_type" default="$(env ROBOT_TYPE)" doc="Robot type: [a1, aliengo, go1, laikago]" />
      <arg name="ns" default="$(env ROBOT_NAME)"/>    
      <arg name="x" default="0" />
      <arg name="y" default="0" />

      <group ns="$(arg ns)">

            <param name="urdfFile" value="/tmp/legged_control/$(arg ns).urdf" />
            <param name="taskFile" value="$(find legged_controllers)/config/$(arg robot_type)/task.info" />
            <param name="referenceFile" value="$(find legged_controllers)/config/$(arg robot_type)/reference.info" />
            <param name="gaitCommandFile" value="$(find legged_controllers)/config/$(arg robot_type)/gait.info" />

            <rosparam file="$(find multi_legged_controllers)/config/controllers.yaml" command="load" />

            <node name="controller_loader" pkg="controller_manager" type="controller_manager"
                  output="screen"
                  args="load
                  controllers/multi_legged_controller
                  controllers/multi_perceptive_controller" />

            <node pkg="legged_deployment" type="gait_command_node" name="gait_command_node"
                  output="screen" />

            <node pkg="legged_deployment" type="CmdPublisher" name="CmdPublisher"
                  output="screen"/>
       </group>     


</launch>